appcelerator/titanium_mobile
# [TIMOB-24851] Android: Deprecated Ti.UI.Label "wordWrap" property

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-24851

**Summary:**
- "wordWrap" property now ignored. Will log deprecation warning when attempting to set it.
- This property is not supported on iOS, making it not portable.
- The equivalent of "wordWrap" set to `false` can be done via "maxLines" set to `1`, which is portable on all platforms.
- See ticket for explanations why this should be deprecated.

**Test:**
1. Build and run the below on Android.
2. Verify that top label shows 2 lines and bottom label only shows 1 line with `...` ellipsis.
3. Verify that the following warning was logged.
`Ti.UI.Label property 'wordWrap' is deprecated. If setting false, then set 'maxLines' to 1 instead.`
4. Comment out the 2 "wordWrap" property lines from the below code.
5. Rebuild and run on Android.
6. Verify that top label shows 2 lines and bottom label only shows 1 line with `...` ellipsis.
7. Verify that a "wordWrap" deprecation warning was **NOT** logged.

```javascript
var window = Ti.UI.createWindow();
window.add(Ti.UI.createLabel({
	text: "This is line 1.\nThis is line 2.",
	wordWrap: true,
	top: "25%",
}));
window.add(Ti.UI.createLabel({
	text: "This is line 1.\nThis is line 2.",
	maxLines: 1,
	wordWrap: false,
	top: "75%",
}));
window.open();
```
